### ISSUE TITLE
Force Log when host/worker is ready for VS Code debugger to attach  when loglevel is set to None

### ISSUE THREAD

**Issue by anthonychu** [Azure/azure-functions-core-tools#2110]
After discussing with @pragnagopa, it sounds like the initial change in Azure/azure-functions-host#6230 is not quite sufficient for the purposes of VS Code attaching the debugger reliably.

There are 2 issues in particular:
- In-proc .NET apps will not print the new message
- The message is printed with log level Information. If default log levels are changed to Warning, Error, or None, the message will not print

We need this message to be printed regardless of stack or log level. Also suggest updating the message to something more generic and doesn't mention "worker process". (something like "Function app is ready"?)

Also needed for this: https://github.com/Azure/azure-functions-core-tools/issues/1131

@ericjizbaMSFT

**Comment by fabiocav** [comment]
Moving this to the Core Tools repo as the work would be done there.

**Comment by anthonychu** [comment]
@pragnagopa Is this issue still needed to track this? From what I can tell, the message is emitted from the host so any changes need to be done there, not Core Tools.

**Comment by pragnagopa** [comment]
> the message is emitted from the host so any changes need to be done there, not Core Tools.

Yes. Log is coming from the host. Both @fabiocav and @brettsam recommended adding a logging provider in core tools that will force log these logging statements even when logging level is set to `None`

I will look into this possibility.

**Comment by pragnagopa** [comment]
@brettsam - I am trying to force logs to flow through if logLevel is set to None and then filter the logs based on the content. Tried the following:

```
.ConfigureLogging( (context, loggingBuilder) =>
                {
                    loggingBuilder.ClearProviders();
                    if (hostJsonLogLevel == LogLevel.None)
                    {
                        loggingBuilder.SetMinimumLevel(LogLevel.Trace);
                        loggingBuilder.AddDefaultWebJobsFilters();
                        loggingBuilder.AddProvider(new ColoredConsoleLoggerProvider(hostJsonLogLevel));
                    }
                    else
                    {
                        loggingBuilder.AddDefaultWebJobsFilters();
                        loggingBuilder.AddProvider(new ColoredConsoleLoggerProvider((cat, level) => level >= LogLevel.Information));
                    }
                })
```
Even though minimum loglevel is set to to `Trace`, only webjost level logs and RpcWorkerChannel logs are written to console.

Any ideas on what I might be missing here?

**Comment by brettsam** [comment]
What logger do you need to write out your log? You need to add a filter specifically for that logger. If you have one, it wins above all other filters, even ones specified in configuration.

For example: `loggingBuilder.AddFilter<ColoredConsoleLoggerProvider>(null, LogLevel.Trace);` 

Setting `null` here means *all* categories will be filtered at `Trace` level. If you want a specific category, you can specify it instead.

See this for how filtering rules work: https://docs.microsoft.com/en-us/aspnet/core/fundamentals/logging/?view=aspnetcore-3.1#how-filtering-rules-are-applied.

**Comment by pragnagopa** [comment]
Moving this issue to Sprint 82 to give more time for PR reviews.

**Comment by fabiocav** [comment]
This is currently in the PR tracking the logging verbosity changes

**Comment by pragnagopa** [comment]
Addressed in referenced PR #2139
